### PR TITLE
[PBE-6015] Add extra info to request client header

### PR DESCRIPTION
**Pull request by liviu-timar**
### 🎯 Goal

Add more info to the `X-Stream-Client` header.

### 🛠 Implementation details

- Added app name to header. Will be provided by client app via new `StreamVideoBuilder` `appName` parameter. Falls back to `packageName` if argument is not provided.
- Added docs for new parameter.
- Removed _duplicated_ header that was added when creating socket. Left just the one added in `HeadersInterceptor#intercept`.

### 🎨 Changes

Example header before:
`stream-video-android-1.0.13|os=Android 14|api_version=34|device_vendor=samsung|device_model=SM-A155F`

Example header now:
`stream-video-android-1.0.13|os=Android 14|api_version=34|device_vendor=samsung|device_model=SM-A155F|app_name=Stream Video Demo App`
or
`stream-video-android-1.0.13|os=Android 14|api_version=34|device_vendor=samsung|device_model=SM-A155F|app_name=io.getstream.video.android.debug`

### CODE REVIEW

File: stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/StreamVideoBuilder.kt [hunk 1]
```diff
@@ -102,6 +103,7 @@ public class StreamVideoBuilder @JvmOverloads constructor(
     private val crashOnMissingPermission: Boolean = false,
     private val permissionCheck: StreamPermissionCheck = DefaultStreamPermissionCheck(),
     private val audioUsage: Int = defaultAudioUsage,
+    private val appName: String? = null,
```
Comment (liviu-timar) [inline stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/StreamVideoBuilder.kt:106]: `appName` is optional. Should we make it mandatory?

Comment (aleksandar-apostolov) [reply stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/StreamVideoBuilder.kt:106]: Can the packagename be the default if not supplied?

File: stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/socket/PersistentSocket.kt [hunk 2]
```diff
@@ -277,7 +276,6 @@ public open class PersistentSocket<T>(
             .url(url)
             .addHeader("Connection", "Upgrade")
             .addHeader("Upgrade", "websocket")
-            .addHeader("X-Stream-Client", StreamVideo.buildSdkTrackingHeaders())
```
Comment (aleksandar-apostolov) [inline stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/socket/PersistentSocket.kt:280]: Why did we remove this?

Comment (liviu-timar) [reply stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/socket/PersistentSocket.kt:280]: The header was duplicated when connecting the socket. It was added here and also in `HeadersInterceptor#intercept`. Tested with this line removed and it's added only once, both in `video/connect` and other requests, like `video/call`.

File: stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/StreamVideoBuilder.kt (hunk 1 above)
Comment (liviu-timar) [reply stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/StreamVideoBuilder.kt:106]: I added the packageName as a fallback. The header when `appName` is not provided looks like: `stream-video-android-1.0.13|os=Android 14|api_version=34|device_vendor=samsung|device_model=SM-A155F|app_name=io.getstream.video.android.debug`

**Review by aleksandar-apostolov — APPROVED** [review]

**Pull request merged**